datafusion/expr/src/literal.rs

```diff
@@ -41,39 +42,45 @@ pub trait TimestampLiteral {
     fn lit_timestamp_nano(&self) -> Expr;
 }
 
+impl From<ScalarValue> for Expr {
+    fn from(value: ScalarValue) -> Self {
+        Self::Literal(Scalar::from(value))
```

Comment:
at some point we should rename Literal to Constant. constant is what this represents, while literal is a syntax-related term.